tests/test_derivative.py

```diff
@@ -8,7 +8,7 @@
 
 
 # defining an observable
-def hamiltonian(nqubits, backend):
+def hamiltonian(nqubits):
     return hamiltonians.hamiltonians.SymbolicHamiltonian(
         np.prod([Z(i) for i in range(nqubits)])
     )
```

Comment:
Doing this will ignore the `backend` parameter in the test and will use the default `GlobalBackend()` for all `test_derivative`. The proper solution would be
```py
def hamiltonian(nqubits, backend):
    return hamiltonians.hamiltonians.SymbolicHamiltonian(
        np.prod([Z(i) for i in range(nqubits)]), backend=backend
    )
```
so that all backends are tested.

Reply:
Implemented